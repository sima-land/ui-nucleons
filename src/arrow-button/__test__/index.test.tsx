import { it, expect, describe, jest } from '@jest/globals';
import { render, fireEvent } from '@testing-library/react';
import { ArrowButton } from '..';

describe('ArrowButton', () => {
  it('should renders correctly', () => {
    const { container } = render(<ArrowButton aria-label='Вперед' />);

    expect(container).toMatchSnapshot();
  });

  it('should handle props', async function () {
    const spy = jest.fn();

    const { container, findByTestId } = render(
      <ArrowButton size='l' direction='left' aria-label='Назад' onClick={spy} />,
    );

    expect(container).toMatchSnapshot();

    expect(spy).toHaveBeenCalledTimes(0);

    fireEvent.click(await findByTestId('arrow-button'));

    expect(spy).toHaveBeenCalledTimes(1);
  });

  it('should handle "data-testid"', function () {
    const { queryAllByTestId } = render(
      <ArrowButton
        size='l'
        direction='left'
        aria-label='Назад'
        data-testid='specific-arrow-button'
      />,
    );

    expect(queryAllByTestId('arrow-button')).toHaveLength(0);
    expect(queryAllByTestId('specific-arrow-button')).toHaveLength(1);
  });
});
