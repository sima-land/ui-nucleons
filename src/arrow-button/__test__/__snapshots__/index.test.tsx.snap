// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowButton should handle props 1`] = `
<div>
  <button
    aria-label="Назад"
    class="arrow-button size-l"
    data-testid="arrow-button"
    type="button"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        clip-rule="evenodd"
        d="M11.707 3.293a1 1 0 0 1 0 1.414L4.414 12l7.293 7.293a1 1 0 0 1-1.414 1.414l-8-8a1 1 0 0 1 0-1.414l8-8a1 1 0 0 1 1.414 0"
        fill-rule="evenodd"
      />
      <path
        clip-rule="evenodd"
        d="M3 12a1 1 0 0 1 1-1h17a1 1 0 1 1 0 2H4a1 1 0 0 1-1-1"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`ArrowButton should renders correctly 1`] = `
<div>
  <button
    aria-label="Вперед"
    class="arrow-button size-l"
    data-testid="arrow-button"
    type="button"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        clip-rule="evenodd"
        d="M11.253 3.336a1 1 0 0 1 1.411-.083l9 8a1 1 0 0 1 0 1.494l-9 8a1 1 0 0 1-1.328-1.494L19.495 12l-8.16-7.253a1 1 0 0 1-.082-1.411"
        fill-rule="evenodd"
      />
      <path
        clip-rule="evenodd"
        d="M2 12a1 1 0 0 1 1-1h17a1 1 0 1 1 0 2H3a1 1 0 0 1-1-1"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;
